src/nvim/fold_defs.h

```diff
@@ -0,0 +1,17 @@
+#ifndef NVIM_FOLD_DEFS_H
```

Comment:
Is creating this really necessary?

Reply:
Why not?

Reply:
I just don't understand the motivation in how it relates to this PR. I don't strongly object to it.

Reply:
I was indeed running into circular includes due to introducing a `foldinfo_T` in the `statuscol_T` struct. The alternative would be (redundatly) calling `fold_info()` again in the statusline item code.